lib/ansible/plugins/lookup/password.py

```diff
@@ -102,6 +104,7 @@
 
 DEFAULT_LENGTH = 20
 VALID_PARAMS = frozenset(('length', 'encrypt', 'chars'))
+LOCKFILE_NAME = to_bytes("ansible_lookup_password.lockfile")
```

Comment:
Doesn't that mean that suddenly, all password lookup are locked on the same file ? If that's for 2 differents password lookup on different file, it should lock. While that's unlikely to happen, I think it would be better to avoid 1 single big lock. Could something like a hash of the file name be used and/or added ?

Reply:
Also, what if a user use a password file with the name "ansible_lookup_password.lockfile"

Reply:
yes make sense, I will find some way to produce a random name so workers will not block on a single lock and the name of lockfile will not be same as password file.

Reply:
Typically it's a convention to have a `.lock` extension for such files.